server/src/test/java/org/apache/druid/server/coordination/SegmentBootstrapperTest.java

```diff
@@ -150,11 +150,18 @@ public void testStartStop() throws Exception
       Assert.assertEquals(2L, segmentManager.getDataSourceCounts().get("test_two" + i).longValue());
     }
 
-    Assert.assertEquals(ImmutableList.copyOf(segments), segmentAnnouncer.getObservedSegments());
-
     final ImmutableList<DataSegment> expectedBootstrapSegments = ImmutableList.copyOf(segments);
-    Assert.assertEquals(expectedBootstrapSegments, cacheManager.getObservedBootstrapSegments());
-    Assert.assertEquals(expectedBootstrapSegments, cacheManager.getObservedBootstrapSegmentsLoadedIntoPageCache());
+
+    // The following verbose assert - seen throughout this test class - confirms list item equality irrespective of item order.
+    Assert.assertTrue(expectedBootstrapSegments.size() == segmentAnnouncer.getObservedSegments().size() &&
+                      expectedBootstrapSegments.containsAll(segmentAnnouncer.getObservedSegments()) &&
+                      segmentAnnouncer.getObservedSegments().containsAll(expectedBootstrapSegments));
```

Comment:
Some suggestions:
- Maybe add a private method `verifyContentsAreSame()` or something in this class to do this check,
and avoid the verbosity.
- A simple set-based comparison would be neater than a two way contains all check.
- Try to avoid asserting multiple conditions in a single assert statement

```java
private static <T> void verifyContentsAreSame(List<T> listA, List<T> listB) {
    Assert.assertEquals(listA.size(), listB.size());
    Assert.assertEquals(Set.copyOf(listA), Set.copyOf(listB));
}
```

Reply:
cool, I appreciate the ideas on cleaning up, I'll do some refactoring. And thanks for the tip on staying away from multi-condition asserts, I'll keep that in mind for future.